appcues/src/main/java/com/appcues/DeepLinkHandler.kt

```diff
@@ -18,36 +20,45 @@ internal class DeepLinkHandler(
     fun handle(activity: Activity, intent: Intent?): Boolean {
         val linkAction: String? = intent?.action
         val linkData: Uri? = intent?.data
-        var handled = false
 
-        if (linkData == null) return handled
+        if (linkData != null) {
+            val validScheme = linkData.scheme == "appcues-${config.applicationId}" || linkData.scheme == "appcues-democues"
+            val validHost = linkData.host == "sdk"
 
-        val validScheme = linkData.scheme == "appcues-${config.applicationId}" || linkData.scheme == "appcues-democues"
-        val validHost = linkData.host == "sdk"
+            if (linkAction == Intent.ACTION_VIEW && validScheme && validHost) {
+                return processLink(linkData, activity)
+            }
+        }
 
-        if (linkAction == Intent.ACTION_VIEW && validScheme && validHost) {
-            val segments = linkData.pathSegments
-            when {
-                segments.count() == 2 && segments[0] == "experience_preview" -> {
-                    appcuesCoroutineScope.launch {
-                        experienceRenderer.preview(segments[1])
-                    }
-                    handled = true
-                }
-                segments.count() == 2 && segments[0] == "experience_content" -> {
-                    appcuesCoroutineScope.launch {
-                        experienceRenderer.show(segments[1], ExperienceTrigger.DeepLink)
-                    }
-                    handled = true
+        return false // link not handled
+    }
+
+    // return true if handled
+    private fun processLink(linkData: Uri, activity: Activity): Boolean {
+        val segments = linkData.pathSegments
+        return when {
+            segments.count() == 2 && segments[0] == "experience_preview" -> {
+                appcuesCoroutineScope.launch {
+                    experienceRenderer.preview(segments[1])
                 }
-                segments.any() && segments[0] == "debugger" -> {
-                    val deepLinkPath = if (segments.count() > 1) segments[1] else null
-                    debuggerManager.start(activity, deepLinkPath)
-                    handled = true
+                true
+            }
+            segments.count() == 2 && segments[0] == "experience_content" -> {
+                appcuesCoroutineScope.launch {
+                    experienceRenderer.show(segments[1], DeepLink)
                 }
+                true
+            }
+            segments.any() && segments[0] == "debugger" -> {
+                val deepLinkPath = if (segments.count() > 1) segments[1] else null
+                debuggerManager.start(activity, Debugger(deepLinkPath))
+                true
             }
+            segments.any() && segments[0] == "capture_screen" -> {
+                debuggerManager.start(activity, ScreenCapture)
+                true
+            }
```

Comment:
This is the new part in this file. The rest is just some restructuring to split things apart and resolve a Detekt warning about too much function complexity (15 branches) in the previous function when this new path was added.